### PR TITLE
sim2h client in hc

### PR DESCRIPTION
**Pull request by zippy**
## PR summary

Adds commands to `hc` that we can use to connect to a sim2h server to get its health status.

## testing/benchmarking notes

( if any manual testing or benchmarking was/should be done, add notes and/or screenshots here )

this needs to be tested manually atm

sim2h_client

- in one tab run `hc-sim2h-server-install && sim2h_server -p 3000`
- in another tab run `hc-cli-install` then:

```
[nix-shell:~/holochain-rust]$ hc sim2h-client -u wss://localhost:3000
url: wss://localhost:3000
message: ping
connecting to: wss://127.0.0.1:3000/
Generated agent id: HcSCiFa3hPh5Ikxnfn4c9Cmq5wjAPddzvGWcFC6kH3Jus6qo7SA6s65Dbvn3xoz
Pong

[nix-shell:~/holochain-rust]$ hc sim2h-client -u wss://localhost:3000 -m status
url: wss://localhost:3000
message: status
connecting to: wss://127.0.0.1:3000/
Generated agent id: HcSCjVagJyrrcvvdgxd8eIbfIp35o3tnren5j5k6ONc5qo7yGc7F7xoSUDpgw5r
StatusResponse(StatusData { spaces: 0, connections: 1 })

[nix-shell:~/holochain-rust]$ hc sim2h-client -u wss://localhost:3000 -m ping
url: wss://localhost:3000
message: ping
connecting to: wss://127.0.0.1:3000/
Generated agent id: HcScIY5J85FMHH4eztyWbkbHx9sb6gnftXC4Rb3kMXqbpta7UaFOkzFf7ve57ir
Pong
```

## followups

( any new tickets/concerns that were discovered or created during this work but aren't in scope for review here )

## changelog

- [ ] if this is a code change that effects some consumer (e.g. zome developers) of holochain core,  then it has been added to [our between-release changelog](https://github.com/holochain/holochain-rust/blob/develop/CHANGELOG-UNRELEASED.md) with the format 

```markdown
- summary of change [PR#1234](https://github.com/holochain/holochain-rust/pull/1234)
```

## documentation

- [ ] this code has been documented according to our [docs checklist](https://hackmd.io/@freesig/Hk9AmKJNS)

### CODE REVIEW

**Comment by Connoropolous** [comment]
what does this do @zippy ?

**Comment by zippy** [comment]
> what does this do @zippy ?

We want a tiny binary that we can run in a cron-job so that we can confirm that a sim2h server is working and accepting connections, and thus if it fails we can restart the sim2h server.

File: crates/sim2h_server/src/bin/sim2h_admin.rs [hunk 1]
```diff
@@ -0,0 +1,187 @@
+//! `cargo run --bin sim2h_admin -- --help`
```
Comment (thedavidmeister) [inline crates/sim2h_server/src/bin/sim2h_admin.rs]: :thinking: adding more binaries adds more ops works, e.g. we now need to get this building and QA'd cross-platform and see if it has any shared dependencies/libs that need to be on people's machines and might need to add it to dockers and need to roll it into a dist/release process that checks shas in holonix, and need to document how people can interact with it without holonix, etc. (e.g. @samcooley chasing up a live sim2h server box going down recently)...

is there a viable "light touch" approach here, e.g. a bash script in holonix that could suffice for a simple web socket health check?

Comment (thedavidmeister) [reply crates/sim2h_server/src/bin/sim2h_admin.rs]: or alternatively, could this be implemented as a `hc` command?

Comment (zippy) [reply crates/sim2h_server/src/bin/sim2h_admin.rs]: yes, lets' move this to hc, that's great.  I've updated the PR title.

**Comment by zippy** [comment]
Nice @thedavidmeister.  I updated the timeout to 1 second after watching it fail when connecting to a live busy server...

**Pull request merged**